Added separator and file_name options

D-K-E opened the pull request:
@kylepjohnson Here are the separator and file_name options we had discussed in issue #408 .
The major problem was maintaining the flow of the word groups while changing file_name column. The solution I could come up with is mapping word groups to tokens and tokens to file names. The rest is, iterating over the word groups to produce the concordance line in relation to the filename associated with the token of the word group.

[Here](https://gist.github.com/D-K-E/5702ffc0456bab090e1db413b88b9a5d) is the concordance before this commit for Philology().write_concordance_from_file()
[Here](https://gist.github.com/D-K-E/e97b3339ac297dd140fdf17b3149383f) is the concordance after this commit for Philology().write_concordance_from_file() default options are changed, meaning: ex.separator="\t", file_names=True
[Here](https://gist.github.com/D-K-E/cefdd45aba8aa762c7212614c786d5f4) is the concordance after this commit for Philology().write_concordance_from_file() default options are in effect, meaning: separator=' ', file_names=False
[Here](https://gist.github.com/D-K-E/2170e371b34049dd7473b8ab25b31aeb) is the concordance after this commit for Philology().write_concordance_from_string(), default options are changed, meaning: ex. separator="\t"
[Here](https://gist.github.com/D-K-E/1bf56d8e1b2d85dc76c6364afa803ca9) is the concordance after this commit for Philology().write_concordance_from_string() default options are in effect, meaning: separator=' '

D-K-E commented:
I'll see what I can do about the codecov test

kylepjohnson commented:
Thanks @D-K-E. This is a little more involved than I anticipated (not a bad thing).

I'll be able to properly review this weekend. I'm unusually busy this week.

D-K-E commented:
@kylepjohnson I tried to understand what is creating the misses from codecov, but I am very unfamiliar with the general code base, so I don't really understand how parts interact with each other. I thought the dict type that I introduced may somehow create an iteration problem so I changed to tuples instead. It increased the coverage a little, but just a little. Would you mind taking a look when you are not busy?

kylepjohnson commented:
Sure, will do. I haven't forgot about this. Thank you for being so patience :)

kylepjohnson commented:
I've looked at this a few times and, to be honest, each time I think this is overkill and not necessary (`re` for example seems like overkill, and it seems like the changes touch too many of the internal functions).
